nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -585,13 +584,18 @@ public Task fetchTaskById(String documentId) {
         return (Task) getResponse(documentId);
     }
 
-    private void refuseIfDateFound(Object obj) {
-        if (obj instanceof Calendar || obj instanceof Date) {
-            throw new IllegalArgumentException(MSG_DATE_UNSUPPORTED);
-        }
-        if (obj instanceof Map) {
-            for (Object value : ((Map) obj).values()) {
-                refuseIfDateFound(value);
+    private void rejectIfDateFound(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
+            throw new IllegalArgumentException(String.format(
+                    "Property '%s' has value of type '%s'. However, date values are not supported in Nuxeo Java Client."
+                            + " Please convert it to String with ISO 8601 format \"yyyy-MM-dd'T'HH:mm:ss.SSSXXX\""
+                            + " before setting it as property.",
+                    key, value.getClass().getTypeName()));
+        } else if (value instanceof Collection && ((Collection) value).iterator().hasNext()) {
+            rejectIfDateFound(key, ((Collection) value).iterator().next());
```

Comment:
In theory it could be `Collection<Object>` and containing both `String` and `Calendar`, so as long as we're checking let's check them all.